pkg/storage/watch_cache.go

```diff
@@ -188,8 +189,16 @@ func (w *watchCache) List() []interface{} {
 	return w.store.List()
 }
 
-func (w *watchCache) ListWithVersion() ([]interface{}, uint64) {
-	w.RLock()
+func (w *watchCache) WaitUntilFreshAndList(resourceVersion uint64) ([]interface{}, uint64) {
+	// TODO: Implement it properly before enabling it in production.
```

Comment:
Yes - I was thinking about it, but I started thinking that if there will be a lot of LISTs each waiting on a different version then we will be broadcasting on every change and waking up all listers every time.
But I guess it's NOT a real problem.

I will fix it tomorrow.